src/frontend/store/shared.ts

```diff
@@ -72,11 +72,40 @@ export const useSharedStore = defineStore('shared', () => {
     hasOpenSidebar.value = isOpen;
   };
 
+  const currentStoryName = ref('');
+
+  // main and header
+
+  const isMainUnderHeader = ref(false);
+
+  const setMainUnderHeader = (isUnder: boolean) => {
+    isMainUnderHeader.value = isUnder;
+  };
+
+  const headerHeight = ref(0);
+  const headerWidth = ref(0);
+  const layoutWidth = ref(0);
+  const containerWidth = ref(0);
+  const setHeaderHeight = (fresh: number) => {
+    headerHeight.value = fresh;
+  };
+
+  const setHeaderWidth = (fresh: number) => {
+    headerWidth.value = fresh;
+  };
```

Comment:
Since we always set them together, how about:

```suggestion
  const setHeaderSize= (height: number, width: number) => {
    headerHeight.value = height;
    headerWidth.value = width;
  };
```